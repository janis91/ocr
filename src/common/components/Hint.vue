<template>
  <div class="ocr-hint">
    <span class="icon icon-details"></span>
    <span>{{hint}}</span>
  </div>
</template>

<script lang="ts">
import Vue from 'vue'

export default Vue.extend({
  name: 'Hint',
  props: {
    hint: {
      type: String,
      required: true
    }
  }
})
</script>

<style scoped>
.ocr-hint {
  padding: 0px 4px;
  color: var(--color-text-light);
}

.ocr-hint >>> span.icon {
  background-size: 16px 16px;
  display: inline-block;
  vertical-align: text-top;
  margin-right: 0.3rem;
}
</style>
