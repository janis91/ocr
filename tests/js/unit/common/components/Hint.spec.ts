import '../../GlobalMocks.mock'
import { mount, Wrapper } from '@vue/test-utils'
import Hint from '@/common/components/Hint.vue'

describe('Hint.vue', () => {
  let wrapper: Wrapper<any>

  beforeEach(() => {
    wrapper = mount(Hint, {
      propsData: {
        hint: 'Test Hint.'
      }
    })
  })

  it('WHEN Hint is mounted, THEN matches snapshot with test text.', async () => {
    expect(wrapper.text()).toContain('Test Hint.')
    expect(wrapper.element).toMatchSnapshot()
  })
})
