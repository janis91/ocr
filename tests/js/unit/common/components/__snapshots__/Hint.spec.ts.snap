// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hint.vue WHEN Hint is mounted, THEN matches snapshot with test text. 1`] = `
<div
  class="ocr-hint"
>
  <span
    class="icon icon-details"
  />
   
  <span>
    Test Hint.
  </span>
</div>
`;
